ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,277 @@
+# 33. Provisioning Clusters for Integration Tests
+
+Date: 2024-03-07
+
+## Status
+
+Accepted
+
+Supersedes:
+
+- [ADR 08. Environment Provisioning](0008-environment-provisioning.html)
+- Environment provisioning parts of [ADR 32. Decoupling Deployment](0032-decoupling-deployment.html)
+
+## Context
+
+This decision clarifies how integration test environments will be dynamically provisioned. In prior
+decisions it was believed that [Dynamic Resource Allocation] (DRA) would graduate out of OpenShift
+TechPreview on a timeline suitable for this project. This is no longer the case and, as such, a new
+approach to requesting compute resources for test pipelines is needed. DRA is still expected to
+become the kubernetes-native approach to managing dynamically provisioned resources across pods.
+Therefore, any interim solution should intentionally avoid introducing new barriers that might
+prevent the adoption of DRA some day.
+
+The problem of provisioning test resources/environments can be broken down into a few questions:
+
+1. How can resources (OpenShift clusters, to start) be provisioned, efficiently, without exposing
+   shared cloud/infra credentials to the end user?
+1. In case the user requires more customization than what's provided by shared configuration, how
+   can they provide their own (including infra credentials)?
+1. How does the end user request resources from an integration `Pipeline`?
+
+Provisioning of infrastructure can consume some significant compute resources itself. All possible
+solutions must also account for the challenge of scalability in production.
+
+### Cluster Provisioning
+
+[Hive] may be used to provision OpenShift clusters. It's widely used in OpenShift CI testing,
+supports hibernated cluster pools for quick (5-7m), cost-efficient, allocation and is maintained
+and distributed by Red Hat on OperatorHub. Architecture support for provisioned clusters is
+determined by the infra provider for which Hive supports all the most popular options. The
+[scaling characteristics][HiveScaling] of Hive on a single cluster are well documented with some
+known upper limits (max of ~1000 provisioned clusters per management cluster).
```

Comment:
Yes, the total number of ephemeral clusters as determined by practical limits on CRD storage (from the Hive scaling link). It would be per management cluster but I don't know what this looks like in terms of number of required workers and instance sizing.